src/api.cpp

```diff
@@ -106,62 +100,62 @@ void sentry_add_breadcrumb(sentry_value_t breadcrumb) {
         return;
     }
 
-    {
-        WITH_LOCKED_SCOPE;
-        g_scope.breadcrumbs.append_bounded(breadcrumb_value,
-                                           SENTRY_BREADCRUMBS_MAX);
-    }
+    Scope::with_scope([breadcrumb_value](Scope &scope) {
+        scope.breadcrumbs.append_bounded(breadcrumb_value,
+                                         SENTRY_BREADCRUMBS_MAX);
+    });
 
     if (g_options->backend) {
         g_options->backend->add_breadcrumb(breadcrumb_value);
     }
 }
 
 void sentry_set_user(sentry_value_t value) {
-    WITH_LOCKED_SCOPE;
-    g_scope.user = Value::consume(value);
+    Scope::with_scope(
+        [value](Scope &scope) { scope.user = Value::consume(value); });
```

Comment:
_nit_: This could be `Scope::with_scope_mut` that then calls `flush_scope`. And the other `Scope::with` could receive a `const Scope &`.